libsolidity/interface/CompilerStack.cpp

```diff
@@ -1218,21 +1223,20 @@ void CompilerStack::generateIR(ContractDefinition const& _contract)
 	if (m_hasError)
 		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Called generateIR with errors."));
 
-	if (!_contract.canBeDeployed())
-		return;
-
-	map<ContractDefinition const*, string const> otherYulSources;
-
 	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
 	if (!compiledContract.yulIR.empty())
 		return;
 
 	string dependenciesSource;
 	for (auto const* dependency: _contract.annotation().contractDependencies)
-	{
 		generateIR(*dependency);
-		otherYulSources.emplace(dependency, m_contracts.at(dependency->fullyQualifiedName()).yulIR);
-	}
+
+	if (!_contract.canBeDeployed())
+		return;
+
+	map<ContractDefinition const*, string const> otherYulSources;
```

Comment:
If one really wants to be nit-pickey about this, then this should actually be a
```suggestion
	map<ContractDefinition const*, string_view const> otherYulSources;
```
:-)...
Not much point in copying them around twice...

Reply:
(i.e. and also passed as such to ``IRGenerator``)